homeassistant/components/emoncms/sensor.py

```diff
@@ -163,6 +166,9 @@ def __init__(
         self._value_template = value_template
         self._attr_native_unit_of_measurement = unit_of_measurement
         self._sensorid = sensorid
+        if uuid is not None:
+            self._attr_unique_id = f'{uuid}-{elem["id"]}'
+            self._attr_name = f"Emoncms{id_for_name}|{elem['tag']}|{elem['name']}"
```

Comment:
I think this name is strange, should we leave the naming for when we migrate to config flow? I think when you are adding `DeviceInfo` you are gaining some benefits there.

Reply:
See https://github.com/home-assistant/core/pull/121336 for a config flow migration of emoncms